### ISSUE TITLE
[URGENT] e.postMessage issue with Safari / Mobile Safari 6.1, 7.2, 8.0

### ISSUE THREAD

**Issue by alexloi** [braintree/braintree-web#98]
Error Output
```
TypeError: 'undefined' is not a function (evaluating 'e.postMessage(JSON.stringify({type:this._names…
```
Sentry detection: 
```
bower_components/braintree-web/dist/braintree.js in ./frame-container at line 4:0
```
Braintree web version: 2.15.4 (latest) - The error existed beforehand as well

We're now seeing this error appear 10-15 p/day, meaning we're probably losing multiple orders a day due to this. We tried debugging on our end, can't find any way to solve this - any ideas?

**Comment by mschinis** [comment]
I have the same issue.

This seems to appear on all iOS versions above 6.0. For some reason the postMessage is undefined and causes a crash. This also appears on all Desktop Safari versions.

Looks like the issue is when the braintree library sends messages between the child iframe (braintree) and the parent frame (website that includes braintree).

**Comment by kyledetella** [comment]
@alexloi sorry to hear you are running into these issues. I assume since your log mentioned `frame-container` you are using a Drop-in integration? Unfortunately, at this point I have not been able to replicate this issue in any of my test apps.

The best way for us to help debug this issue is to take a look at your integration. I would reach out to our support team (support@braintreepayments.com) and provide a link to your integration and relevant `braintree.setup` code.

@mschinis is your issue also with Drop-in? If so, could you pass along your info to support as well?

**Comment by alexloi** [comment]
@kyledetella correct, we're using the drop-in. Will drop you an email now, thanks

**Comment by mschinis** [comment]
Hello @kyledetella, thanks for the swift reply. Yeah, using the drop-in UI. Will reach out to support for help. Thanks!

**Comment by kyledetella** [comment]
@alexloi @mschinis we just released version [`2.16.1`](https://github.com/braintree/braintree-web/tree/2.16.1) which includes a fix to address this issue.

Let us know if you run into anything else.

**Comment by mschinis** [comment]
Thanks. I will be monitoring it over the next couple of days and post here if the issue comes up again.
